src/main/java/org/tugraz/sysds/runtime/controlprogram/context/SparkExecutionContext.java

```diff
@@ -714,11 +714,79 @@ else if(inputInfo2 == InputInfo.BinaryCellInputInfo) {
 		return bret;
 	}
 
+	@SuppressWarnings("unchecked")
+	public PartitionedBroadcast<TensorBlock> getBroadcastForTensorObject(TensorObject to) {
+		long t0 = DMLScript.STATISTICS ? System.nanoTime() : 0;
+
+		PartitionedBroadcast<TensorBlock> bret = null;
+
+		//reuse existing broadcast handle
+		if (to.getBroadcastHandle() != null && to.getBroadcastHandle().isPartitionedBroadcastValid()) {
+			bret = to.getBroadcastHandle().getPartitionedBroadcast();
+		}
+
+		//create new broadcast handle (never created, evicted)
+		if (bret == null) {
+			//account for overwritten invalid broadcast (e.g., evicted)
+			if (to.getBroadcastHandle() != null)
+				CacheableData.addBroadcastSize(-to.getBroadcastHandle().getPartitionedBroadcastSize());
+
+			//obtain meta data for matrix
+			DataCharacteristics dc = to.getDataCharacteristics();
+			long[] dims = new long[dc.getNumDims()];
+			int[] blen = new int[dc.getNumDims()];
+			for (int i = 0; i < dc.getNumDims(); i++) {
+				dims[i] = dc.getDim(i);
+				blen[i] = (int) dc.getBlockSize(i);
+			}
+
+			//create partitioned matrix block and release memory consumed by input
+			TensorBlock tb = to.acquireRead();
+			PartitionedBlock<TensorBlock> pmb = new PartitionedBlock<>(tb, dims, blen);
+			to.release();
```

Comment:
You can say directly acquireReadAndRelease as you anyway hold a strong reference to tb here.